torchvision/ops/giou_loss.py

```diff
@@ -1,4 +1,13 @@
 import torch
+from torch import Tensor
+
+
+def _upcast(t: Tensor) -> Tensor:
+    # Protects from numerical overflows in multiplications by upcasting to the equivalent higher type
+    if t.is_floating_point():
+        return t if t.dtype in (torch.float32, torch.float64) else t.float()
+    else:
+        return t if t.dtype in (torch.int32, torch.int64) else t.int()
```

Comment:
This method already exists at [boxes.py](https://github.com/pytorch/vision/blob/289fce29b3e2392114aadbe7a419df0f2e3ac1be/torchvision/ops/boxes.py#L218-L223) but I'm not sure we should use it here. This is because it keeps boxes in integer float and that would break giou's `mean()` reduction. Instead we should cast to float if not already float32/float64.

Reply:
@datumbox 

Thanks for your review!

Should we newly define upcast method for this function like the below code?

```python
if t.dtype not in (torch.float32, torch.float64):
 return t.float()
```

Reply:
A private method that does that would be fine. No strong opinions. :)

Reply:
I modified it!

https://github.com/pytorch/vision/blob/dc2d754410e25557edc233c54b99b9f3247684fb/torchvision/ops/giou_loss.py#L5-L9